src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Win32Unix.cs

```diff
@@ -51,11 +52,33 @@ internal static void GetDefaultIdentityAndHash(out object identity, out string h
             Assembly? assembly = Assembly.GetEntryAssembly();
 
             if (assembly == null)
+            {
                 throw new IsolatedStorageException(SR.IsolatedStorage_Init);
+            }
+
+            // in case of SingleFile, Location.Length is 0, we will use native entrypoint's filename.
+            if (assembly.Location.Length == 0)
+            {
+                using (Process currentProcess = Process.GetCurrentProcess())
+                {
+                    string? processName = currentProcess.MainModule?.FileName;
+
+                    if (processName == null)
+                    {
+                        throw new IsolatedStorageException(SR.IsolatedStorage_Init);
+                    }
+
+                    var processUri = new Uri(processName);
+                    hash = "Url" + separator + IdentityHelper.GetNormalizedUriHash(processUri);
+                    identity = processUri;
+                }
+
+                return;
+            }
 
             AssemblyName assemblyName = assembly.GetName();
 #pragma warning disable SYSLIB0012
-            Uri codeBase = new Uri(assembly.CodeBase!);
+            var codeBase = new Uri(assembly.CodeBase!);
```

Comment:
nit: Please don't add var to the codebase that is not using it, even if its use is permitted.

Reply:
@ericstj I thought it was ok to use `var` when an object was created via `new` and a constructor. Has this guidance changed? I'm not sure if it's documented.

Reply:
That's true, however usually I try to avoid adding the use of var into a file/codebase that never had it before.  Especially for a "legacy" codebase like this.

Reply:
`#10` clearly permits it: https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/coding-style.md. if this is an exception to the rule, we can update the doc. My muscle memory led me to use `var` here; since RHS is `new`ing `Uri`.